src/management-system-v2/lib/data/users.tsx

```diff
@@ -0,0 +1,74 @@
+'use server';
+
+import { getCurrentUser } from '@/components/auth';
+import {
+  deleteUser as _deleteUser,
+  updateUser as _updateUser,
+  usersMetaObject,
+  UserHasToDeleteOrganizationsError,
+} from './legacy/iam/users';
+import { userError } from '../user-error';
+import { AuthenticatedUserData, AuthenticatedUserDataSchema } from './user-schema';
+import { ReactNode } from 'react';
+import { getEnvironmentById } from './legacy/iam/environments';
+import { OrganizationEnvironment } from './environment-schema';
+import Link from 'next/link';
+
+export async function deleteUser() {
+  const { userId } = await getCurrentUser();
+
+  try {
+    _deleteUser(userId);
+  } catch (e) {
+    let message: ReactNode = 'Error deleting user';
+
+    if (e instanceof UserHasToDeleteOrganizationsError) {
+      const conflictingOrgsNames = e.conflictingOrgs.map(
+        (orgId) => (getEnvironmentById(orgId) as OrganizationEnvironment).name,
+      );
+
+      message = (
+        <>
+          <p>
+            You're part of organizations where you are the only admin, in order to delete your
+            accounts you first have to either add another admin or delete these organizations.
+          </p>
+          <p>The affected organizations are:</p>
+          <ul>
+            {conflictingOrgsNames.map((name) => (
+              <li>{name}</li>
+            ))}
+          </ul>
+          <p>
+            You can delete these organizations <Link href="/environments">here</Link>.
```

Comment:
I would also mention again that the user can add new admins here, so this is not the only CTA to solve it. Or just make it more generic like "You can manage your spaces here".

Reply:
Makes sense